src/request.ts

```diff
@@ -244,6 +244,21 @@ export async function prepareXchainRequestFulfillmentTransaction({
 		toAddress: recipientAddress,
 		enableBoost: true,
 	})
+
+	// Transaction estimation from Squid API allows us to know the transaction fees (gas and fee), then we can iterate over them and add the values ​​that are in dollars
+	let feeEstimation = 0
+	if (routeResult.txEstimation.feeCosts.length > 0) {
+		routeResult.txEstimation.feeCosts.forEach((fee) => {
+			feeEstimation += Number(fee.amountUsd)
+		})
+	}
+
+	if (routeResult.txEstimation.gasCosts.length > 0) {
+		routeResult.txEstimation.gasCosts.forEach((gas) => {
+			feeEstimation += Number(gas.amountUsd)
+		})
+	}
+
```

Comment:
@pawell24 when i say add a comment, i mean on what the actual things are doing. Why are we looping TWICE? what is the difference between gasCosts and FeeCosts? Whhy are we adding BOTH of them to feeEstaimation? Why is there so many gasCosts? Why does it have a length?